packages/opentelemetry-metrics/src/export/Controller.ts

```diff
@@ -38,12 +38,19 @@ export class PushController extends Controller {
     unrefTimer(this._timer);
   }
 
-  private _collect() {
-    this._meter.collect();
-    this._exporter.export(this._meter.getBatcher().checkPointSet(), result => {
-      if (result !== ExportResult.SUCCESS) {
-        // @todo: log error
+  async shutdown(): Promise<void> {
+    await this._collect();
+  }
+
+  private async _collect() {
+    await this._meter.collect();
+    await this._exporter.export(
```

Comment:
"export" function from exporters return "void" there is no reason to call "await" in front of it as it will not block anything.
IF you really want to wait for exporter to finish you should create here a promise and resolve or reject it in a callback where you added "@todo

Reply:
The todo was actually there before my time, I removed this await - you make a good point about it not being async.

Reply:
but you can return a promise to be resolved when exporter exports something (or rejected depends on callback) so this way the `_collect` will really wait until the exporter finishes - what was the original plan I believe